### ISSUE TITLE
The isMoving verification on the client is false even when walking.

### ISSUE THREAD

**Issue by blinkuz** [AscensionGameDev/Intersect-Engine#593]
**Describe the bug**
When using a custom movement animation that depends on the verification that isMoving = true they do not work correctly since the following condition returns false even when moving, which causes frame 1 of the normal spritesheet to appear in the middle of the movement animation, condition:

https://github.com/AscensionGameDev/Intersect-Engine/blob/b408676dd0ddf808b860391d8186c0c825f6a81c/Intersect.Client/Entities/Entity.cs#L601

**Version Encountered**

Prerelease 7 and I believe all previous versions.

**To Reproduce**

There are several ways the easiest would be this:

1. Add Console.WriteLine("IsMoving False!");inside the check for  offsetX and offsetY mentioned in the bug description.
2. Open the client in debug mode and open the debugging console.
3. Start walking and look at the console where you will see that it will give false even if you are moving.

The visual way to reproduce it is:

1. Add new custom walking spritesheet, example:
https://s3.us-east-2.amazonaws.com/ascensiongamedev/filehost/e96a657172e8a005b88524da3110f844.png
2. Start moving and see how the sprite of the normal spritesheet appears suddenly in the middle of its movement animation giving the impression of a very annoying little tick.

**Expected behavior**
The only thing I can think of is to add an extra condition to make sure the character is in motion or check the current conditions related to the offset so that the variable does not change.

Something that occurs to me is that maybe by using the character's speed the change between being still and starting to walk can be taken into account.

In conclusion, the expected behavior is a correct assignment of the variable isMoving.

**Operating System:**
Windows.

**Comment by jcsnider** [comment]
I don't see this as a bug. Technically movement is a series of steps. The character takes a step, stops moving, and then checks the input for directional keys the following loop and if pressed the character starts moving again. When that order of events is done quickly it looks like seamless movement when in fact it is not.

Even if you rearrange the Player.Update() function so processing input happens immediately after the base.Entity call there will still be times in between movements where IsMoving is false, which is technically correct.

For your extra animation you will have to come up with your own way to determine if the sprite is in a sequence of movements. I'd keep a timer for when the most recent step ended (say within 20ms), and then check if the player IsBusy, if the Movement keys are still down, and if the tile in that direction is clear. If all of those conditions are true then I'd keep using the walking anim instead of the normal one.
